azure/services/scalesets/scalesets.go

```diff
@@ -537,6 +537,16 @@ func (s *Service) generateStorageProfile(vmssSpec azure.ScaleSetSpec, sku resour
 			Lun:          disk.Lun,
 			Name:         to.StringPtr(azure.GenerateDataDiskName(vmssSpec.Name, disk.NameSuffix)),
 		}
+
+		if disk.ManagedDisk != nil {
```

Comment:
can we add a test for this or at least modify an existing test to check if the `storageProfile` values are set properly?